dali/operators/generic/resize/tensor_resize.h

```diff
@@ -0,0 +1,206 @@
+  int FirstSpatialDim() const { return first_spatial_dim_; }
+
+  bool CanInferOutputs() const override { return true; }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override;
+
+  void RunImpl(Workspace &ws) override;
+
+  void TrimSpatialDims(const TensorListShape<> &input_shape, int min_ndim) {
+    int nsamples = input_shape.num_samples();
+    auto unchanged_dim = [&](int d) {
+      for (int i = 0; i < nsamples; i++) {
+        int64_t extent = input_shape.tensor_shape_span(i)[d];
+        if (static_cast<int64_t>(resize_params_[i].dst_size[d]) != extent ||
+            static_cast<int64_t>(resize_params_[i].src_lo[d]) != 0 ||
+            static_cast<int64_t>(resize_params_[i].src_hi[d]) != extent) {
+          return false;
+        }
+      }
+      return true;
+    };
+
+    int can_trim_n = std::max(0, spatial_ndim_ - min_ndim);  // at least 2 spatial dims should remain
+    int new_first_spatial_dim = first_spatial_dim_;
+    int new_end_spatial_dim = first_spatial_dim_ + spatial_ndim_;
+
+    for (int d = first_spatial_dim_; d < first_spatial_dim_ + spatial_ndim_; d++) {
+      if (can_trim_n == 0 || !unchanged_dim(d))
+        break;
+      new_first_spatial_dim++;
+      can_trim_n--;
+    }
+    for (int d = new_end_spatial_dim - 1; d > new_first_spatial_dim; d--) {
+      if (can_trim_n == 0 || !unchanged_dim(d))
+        break;
+      new_end_spatial_dim++;
+      can_trim_n--;
+    }
+
+    int new_spatial_ndim = new_end_spatial_dim - new_first_spatial_dim;
+    if (first_spatial_dim_ == new_first_spatial_dim && spatial_ndim_ == new_spatial_ndim)
+      return;
+
+    for (int s = 0; s < nsamples; s++) {
+      auto &p = resize_params_[s];
+      for (int d = 0; d < new_spatial_ndim ; d++) {
+        int orig_d =  new_first_spatial_dim - first_spatial_dim_ + d;
+        p.dst_size[d] = p.dst_size[orig_d];
+        p.src_hi[d] = p.src_hi[orig_d];
+        p.src_lo[d] = p.src_lo[orig_d];
+      }
+      p.dst_size.resize(new_spatial_ndim);
+      p.src_hi.resize(new_spatial_ndim);
+      p.src_lo.resize(new_spatial_ndim);
+    }
+    first_spatial_dim_ = new_first_spatial_dim;
+    spatial_ndim_ = new_spatial_ndim;
+  }
+
+  void PrepareParams(const ArgumentWorkspace &ws, const TensorListShape<> &input_shape,
+                     const TensorLayout &layout) {
+    resize_attr_.PrepareResizeParams(spec_, ws, input_shape, layout);
+
+    // 1st: Expand number of dimensions to at least 2D
+    int min_spatial_ndim = 2;  // ResizeBase requires it.
+    int orig_spatial_ndim = resize_attr_.NumSpatialDims();
+    add_leading_spatial_ndim_ = std::max(0, min_spatial_ndim - orig_spatial_ndim);
+    first_spatial_dim_ = 0;
+    spatial_ndim_ = orig_spatial_ndim + add_leading_spatial_ndim_;
+    expand_dims(expanded_input_shape_, input_shape, first_spatial_dim_, spatial_ndim_);
+    (void) input_shape;  // should use expanded_input_shape_ from now on
```

Comment:
What is this line for? Does the compiler issue a warning when a variable is used after a void cast or sth? If not, leave just the comment.

Reply:
it was more for "documentation", but I can remove it.